whois-api/src/test/java/net/ripe/db/whois/api/mail/dequeue/BounceMailBounceTestIntegration.java

```diff
@@ -0,0 +1,228 @@
+        assertThat(notifyMessageId, Matchers.is(not(nullValue())));
+
+        // Generate failure response for notification
+        insertOutgoingMessageId("XXXXXXXX-5AE3-4C58-8E3F-860327BA955D@ripe.net", "nonexistant@host.org");
+        final MimeMessage message = MimeMessageProvider.getUpdateMessage("permanentFailureMessageRfc822.mail");
+        insertIncomingMessage(message);
+
+        // Wait for address to be marked as undeliverable
+        Awaitility.waitAtMost(10L, TimeUnit.SECONDS).until(() -> (isUndeliverableAddress("nonexistant@host.org")));
+    }
+
+    @Test
+    public void do_send_notification_mail_to_deliverable_address() throws Exception {
+        final String role =
+                "role:        dummy role\n" +
+                "address:       Singel 258\n" +
+                "e-mail:        dummyrole@ripe.net\n" +
+                "phone:         +31 6 12345678\n" +
+                "notify:        Non Existant <nonexistant@ripe.net>\n" +
+                "nic-hdl:       DR1-TEST\n" +
+                "mnt-by:        OWNER-MNT\n" +
+                "source:        TEST\n";
+
+        // send message to mailupdates
+        final String from = insertIncomingMessage("NEW", role + "\npassword: test\n");
+
+        // read reply from mailupdates
+        final MimeMessage acknowledgement = mailSenderStub.getMessage(from);
+        assertThat(acknowledgement.getContent().toString(), containsString("Create SUCCEEDED: [role] DR1-TEST   dummy role"));
+
+        // make sure that normalised email address is stored in outgoing messages table
+        assertThat(countOutgoingForAddress("nonexistant@ripe.net"), is(1));
+
+        // test that there is a notification mail sent to nonexistant@ripe.net
+        assertThat(mailSenderStub.anyMoreMessages(), is(true));
+    }
+
+    @Test
+    public void dont_send_notification_mail_to_undeliverable_address() throws Exception {
+        final String role =
+                "role:        dummy role\n" +
+                "address:       Singel 258\n" +
+                "e-mail:        dummyrole@ripe.net\n" +
+                "phone:         +31 6 12345678\n" +
+                "notify:        nonexistant@ripe.net\n" +
+                "nic-hdl:       DR1-TEST\n" +
+                "mnt-by:        OWNER-MNT\n" +
+                "source:        TEST\n";
+
+        // mark address as undeliverable
+        insertUndeliverableAddress("nonexistant@ripe.net");
+
+        // send message to mailupdates
+        final String from = insertIncomingMessage("NEW", role + "\npassword: test\n");
+
+        // read reply from mailupdates
+        final MimeMessage acknowledgement = mailSenderStub.getMessage(from);
+        assertThat(acknowledgement.getContent().toString(), containsString("Create SUCCEEDED: [role] DR1-TEST   dummy role"));
+
+        // test that no notification mail is sent to nonexistant@ripe.net
+        assertThat(mailSenderStub.anyMoreMessages(), is(false));
```

Comment:
Assertion should be TRUE (we should review all tests)

Reply:
why should be true?
// test that no notification mail is sent to nonexistant@ripe.net

this user is blocked, we shouldn't receive any mail

Reply:
The test "any more messages == false" implies there *are* more message(s)